fud2/src/lib.rs

```diff
@@ -151,11 +148,56 @@ pub fn build_driver(bld: &mut DriverBuilder) {
         Ok(())
     });
 
+    // Creating custom testbench
+    let verilog_refmem = bld.state("verilog-refmem", &["sv"]);
+    let verilog_refmem_noverify = bld.state("verilog-refmem-noverify", &["sv"]);
+
     // Icarus Verilog.
     let verilog_noverify = bld.state("verilog-noverify", &["sv"]);
     let icarus_setup = bld.setup("Icarus Verilog", |e| {
         e.var("iverilog", "iverilog")?;
-        e.rule("icarus-compile", "$iverilog -g2012 -o $out tb.sv $in")?;
+        e.rule(
+            "icarus-compile",
+            "$iverilog -g2012 -o $out tb.sv $additional-input $in",
+        )?;
+        Ok(())
+    });
+    // [Should be default] Setup for using rsrc/tb.sv as testbench (and managing memories within the design)
```

Comment:
What does "should be the default" mean here? Seems like perhaps it is already the default?

Reply:
From my understanding fud2 nondeterministically chooses a path from the source to the destination, and unless you specify with `--through`, there is no "default" path chosen by fud2. My concern was that if someone ran something like `fud2 <CALYX_PROGRAM> --to dat -s sim.data=...`, then fud2 may choose to go through the FIRRTL backend (in which case we need YXI... but if the user didn't build Calyx with YXI enabled then fud2 will crash on them). So that's why I said "should be". Do you think there's a way to circumvent this potential problem?

Although, this may not be a worthwhile problem trying to fix, especially if we are moving in a direction where we want to always use the custom testbench (and therefore always build YXI).

Reply:
Ah great, thanks for explaining! Yeah, FWIW, I think the search proceeds "left to right," i.e., the first-registered operation wins. I like the idea of having it specified this way because it allows for a kind of implicit priority. So anyway, sounds good; I think the default is indeed where we want it to be already.

Reply:
Ah gotcha! Then we should be good :) Thanks!!